doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+      order : string, 'none' or 'canonical'
+          The order by which Mul and Add arguments are processed. For large
+          expressions where speed is a concern, use the setting order='none'.
+
+      """
+
+5. See Also Section
+-------------------
+
+This section is encouraged. If you choose to include a See Also section in your
+docstring, it should be labeled with the heading “See Also” underlined with the
+same length in equals signs.
+
+::
+
+  See Also
+  ========
+
+This section consists of a listing of related functions, classes, and methods.
+The related items can be described with a concise fragment (not a full
+sentence) if desired, but this is not required. If the description spans more
+than one line, subsequent lines must be indented.
+
+Do not reference classes with ``class:Classname``, ``class:`Classname```, or
+``:class:`Classname```, but rather only by their name if they are in the same
+file as the See Also section, or by the full path if they are in a different
+file. Either way, the names listed in the See Also section should actually link
+to the other functions. (WORK IN PROGRESS via issue `#17619
+<https://github.com/sympy/sympy/issues/17619>`_: provide a fix so that any name
+from the SymPy namespace links automatically so you do not have to use the full
+path even if the function is in a different file.)
+
+**Examples**
+
+Here is a correctly formatted See Also section with concise descriptions::
+
+  class erf(Function):
+      r"""
+      The Gauss error function.
+
+      See Also
+      ========
+
+      erfc: Complementary error function.
+      erfi: Imaginary error function.
+      erf2: Two-argument error function.
+      erfinv: Inverse error function.
+      erfcinv: Inverse Complementary error function.
+      erf2inv: Inverse two-argument error function.
+
+      """
+
+Here is a correctly formatted See Also section with just a list of names::
+
+  class besselj(BesselBase):
+      r"""
+      Bessel function of the first kind.
+
+      See Also
+      ========
+
+      bessely, besseli, besselk
+
+      """
+
+6. References Section
+---------------------
+
+This section is encouraged. If you choose to include a References section in
+your docstring, it should be labeled with the heading “References” underlined
+with the same length in equals signs.
+
+::
+
+  References
+  ==========
+
+This section consists of a list of references cited anywhere in the previous
+sections.
+
+The References section should include links to Wikipedia, Wolfram MathWorld,
+DLMF, paper citations, and/or any other resources giving general information
+about the function. References are meant to augment the docstring, but should
+not be required to understand it. References are numbered, starting from one,
+in the order in which they are cited.
+
+References for papers should include, in this order: reference citation, author
+name, title of work, journal or publication, year published, page number.
+
```

Comment:
If there is a DOI, include it in the citation and make sure it is a clickable hyperlink.